src/network.erl

```diff
@@ -25,13 +25,12 @@ request(ApiKey, Endpoint, Input) ->
     jiffy:encode(
       maps:put(<<"statsigMetadata">>, utils:get_statsig_metadata(), Input)
     ),
-  case hackney:post(Api ++ Endpoint, Headers, RequestBody, []) of
-    {ok, StatusCode, _RespHeaders, ClientRef} ->
+
+  case hackney_client:request(post, Api ++ Endpoint, RequestBody, RequestHeaders) of
```

Comment:
how do I set a default client for the application and call the generic behavior method? This is still hard coded to the hackney client

Reply:
You should be able to pass the module as an argument, e.g. `ClientModule`, and then call
`ClientModule:request(post...)`

Reply:
Ended up with `ClientModule = application:get_env(statsig, http_client, hackney_client),`